lib/Serge/Engine/Plugin/parse_yaml.pm

```diff
@@ -134,12 +141,30 @@ sub parse {
         die $error_text;
     }
 
+    if ($self->{data}->{yaml_mode} eq 'rails') {
+        my @tree_keys = keys(%$tree);
+        my $tree_count = int(@tree_keys);
+        if ($tree_count == 0) {
+            # Special case NOOP. Empty data
+        } elsif ($tree_count == 1) {
+            $tree = $tree->{$tree_keys[0]};
+        } else {
+            my $err = join(', ', @tree_keys);
+            $err = "YAML file with more than 1 key at root in yaml_mode: rails: $err";
+            $self->{errors}->{$self->{parent}->{engine}->{current_file_rel}} = $err;
+            # Or should we die?
+            warn "WARNING: $err";
+        }
+    }
+
     # Process tree recursively
 
     $self->process_node('', $tree, $callbackref, $lang);
 
     # Reconstruct YAML
 
+    $tree = {$lang => $tree} if ($lang && $self->{data}->{yaml_mode} eq 'rails');
```

Comment:
If Rails allows you to write any language (like `pt-br`, `pt-BR` or `pt_BR`), I'd suggest passing the language through  `output_lang_rewrite` mapping and output it as is. This gives one a full control on how the languages should be presented in the file with one caveat that the same mapping would be used for file names as well.

Consider this example: you have `ar` language in the list of destination languages, you can use `output_lang_rewrite` to map it to `ar-AE`, and then this value will both be exposed as a root node and as a part of the filename (e.g. `ar-AE.yml`). But I guess this is what a Rails app would expect, anyway.